cmd/auth-rest/startcmd/start.go

```diff
@@ -78,6 +89,31 @@ const (
 	invalidDatabaseTypeErrMsg = "%s is not a valid database type. Run start --help to see the available options"
 )
 
+// OIDC parameters.
+const (
+	// assumed to be the same landing page for all callbacks from all OIDC providers
+	oidcCallbackURLFlagName  = "oidcCallbackURL"
+	oidcCallbackURLFlagUsage = "Base URL for the OIDC callback endpoint." +
+		" Alternatively, this can be set with the following environment variable: " + oidcCallbackURLEnvKey
+	oidcCallbackURLEnvKey = "AUTH_REST_OIDC_CALLBACK"
+
+	googleProviderFlagName  = "googleURL"
```

Comment:
i was working on creating an configuration api .. where we can just call the api instead of passing these variables one at time in the start up.  Api will have two endpoints for now, /create to create the oidc client and /get and then these endpoint will be served in the ui. That we can add any number of clients

Reply:
@talwinder50 where will the secrets be stored? hub-auth should not store them

Reply:
@talwinder50 also I think you meant "any number of providers"

Reply:
> @talwinder50 where will the secrets be stored? hub-auth should not store them

That part need discussion.Right now, was just thinking to use mock oidc provider only

Reply:
We can merge this cut for now and then i will clean up once i raise the configure oidc client api pr by tomorrow.

Reply:
@talwinder50 

> > @talwinder50 where will the secrets be stored? hub-auth should not store them
> 
> That part need discussion.Right now, was just thinking to use mock oidc provider only

This PR adds a mock OIDC provider... please check it out.